src/app/debug/filter-side-drawer/filter-side-drawer.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FilterService } from './filter.service';
+import { Subscription } from 'rxjs';
+import { animate, style, transition, trigger } from '@angular/animations';
+import { auto } from '@popperjs/core';
+
+@Component({
+  selector: 'app-filter-side-drawer',
+  templateUrl: './filter-side-drawer.component.html',
+  styleUrl: './filter-side-drawer.component.css',
+  animations: [
+    trigger('removeTrigger', [
+      transition(':enter', [
+        style({ transform: 'translateX(100%)' }),
+        animate('300ms ease-in', style({ transform: 'translateX(0)' })),
+      ]),
+      transition(':leave', animate('300ms ease-out', style({ transform: 'translateX(100%)' }))),
+    ]),
+  ],
+})
+export class FilterSideDrawerComponent implements OnDestroy, OnInit {
+  protected shouldShowFilter!: boolean;
+  shouldShowFilterSubscriber!: Subscription;
+  protected metadataNames!: string[];
+  metadataNamesSubscriber!: Subscription;
+  protected filters: Record<string, string> = {};
+  filterSubscriber!: Subscription;
```

Comment:
I think its a bit nicer to order the class members by accessibility

```suggestion
  protected shouldShowFilter!: boolean;
  protected metadataNames!: string[];
  protected filters: Record<string, string> = {};

  shouldShowFilterSubscriber!: Subscription;
  metadataNamesSubscriber!: Subscription;
  filterSubscriber!: Subscription;
```

Reply:
yeah that makes sense, I'll add the change